### PR TITLE
Add mutual information to the Table Accessor

### LINKED ISSUE #558: Implement df.ww.mutual_information
- Should include the check to not calculate mutual info on unique columns

### PR DESCRIPTION
**Pull request by tamargrey**
- Adds ability to calculate mutual information for the DataFrame
- Closes #558

### CODE REVIEW

File: woodwork/table_accessor.py [hunk 1]
```diff
@@ -232,6 +241,147 @@ def _get_subset_df_with_schema(self, cols_to_include):
...
+        """Transforms dataframe columns into numeric categories so that
+        mutual information can be calculated
+
+        Args:
+            data (pd.DataFrame): dataframe to use for caculating mutual information
+            num_bins (int): Determines number of bins to use for converting
+                numeric features into categorical.
+
+        Returns:
+            data (pd.DataFrame): Transformed data
```
Comment (thehomebrewnerd) [inline woodwork/table_accessor.py:280]: We should update this to match the style used in `_replace_nans_for_mutual_info` -> `pd.DataFrame: Transformed data`?

File: woodwork/table_accessor.py [hunk 2]
```diff
@@ -232,6 +241,147 @@ def _get_subset_df_with_schema(self, cols_to_include):
...
+        """
+
+        col_names = data.columns.to_list()
```
Comment (thehomebrewnerd) [inline woodwork/table_accessor.py:283]: Do we need this list conversion, or can we just loop over data.columns in the next line -> `for col_name in data.columns:`?

**Review by thehomebrewnerd — COMMENTED** [review]
Overall looks pretty good.

Topic for another day/PR: Should we move the methods `mutual_information_dict`, `_make_categorical_fo_mutual_info` and `_replace_nans_for_mutual_info` to a separate utils file? Not sure these need to reside on the accessor, but we would just have to update them to accept the column info as an additional parameter.

File: woodwork/table_accessor.py (hunk 1 above)
Comment (tamargrey) [reply woodwork/table_accessor.py:280]: Yeah, that makes sense to me

File: woodwork/table_accessor.py (hunk 2 above)
Comment (tamargrey) [reply woodwork/table_accessor.py:283]: Doesn't seem like we use the list of column names anywhere, so happy to make this change

File: woodwork/utils.py [hunk 3]
```diff
@@ -136,7 +136,7 @@ def read_csv_to_accessor(filepath=None,
             information on available keywords refer to the pandas documentation.
 
     Returns:
-        woodwork.DataTable: DataTable created from the specified CSV file
+        pd.DataFrame: DataFrame created from the specified CSV file with Woodwork typing information initialized.
```
Comment (tamargrey) [inline woodwork/utils.py:139]: realized this docstring was still referencing DataTables so I changed it

**Comment by tamargrey** [comment]
@thehomebrewnerd regarding moving mutual_information_dict, _make_categorical_fo_mutual_info and _replace_nans_for_mutual_info into a separate utils file, I think that's not a bad idea. Could be `statistics/utils` or something to go along with the `test_statitstics` file. 

If we're happy with that, I can start with `describe` when working on that issue, and move the mutual info stuff after.

**Comment by thehomebrewnerd** [comment]
> @thehomebrewnerd regarding moving mutual_information_dict, _make_categorical_fo_mutual_info and _replace_nans_for_mutual_info into a separate utils file, I think that's not a bad idea. Could be `statistics/utils` or something to go along with the `test_statitstics` file.
> 
> If we're happy with that, I can start with `describe` when working on that issue, and move the mutual info stuff after.

Sounds good to me.

**Review by thehomebrewnerd — APPROVED** [review]
Looks good!

**Pull request merged**